web/src/components/lineage/Lineage.tsx

```diff
@@ -126,10 +134,10 @@ export function getSelectedPaths(g: graphlib.Graph<MqNode>, selectedNode: string
 
     const predecessors = g?.predecessors(node)
     if (predecessors?.length) {
-      for (let i = 0; i < node.length - 1; i++) {
+      for (let i = 0; i < predecessors.length; i++) {
         if (predecessors[i]) {
-          paths.push([predecessors[i] as unknown as string, node])
-          getPredecessors(predecessors[i] as unknown as string)
+          paths.push([(predecessors[i] as unknown) as string, node])
+          getPredecessors((predecessors[i] as unknown) as string)
```

Comment:
Not sure how long we've been doing this but it looks like we've been iterating over the characters in the node id 😬 ? Update this to traverse the predecessors array and did the same for getSuccessors